### ISSUE TITLE
profiling extension install error

### ISSUE THREAD

**Issue by zach-taylor** [DataDog/dd-trace-rb#2431]
**Current behaviour**
`ddtrace` gem is failing to install in our Rails app bundle install

**Expected behaviour**
`ddtrace` gem should install

**Steps to reproduce**

See gem/library versions below

Install error message:

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
--
  |  
  | current directory:
  | /app/vendor/bundle/ruby/2.7.0/gems/ddtrace-1.6.1/ext/ddtrace_profiling_native_extension
  | /usr/local/bin/ruby -I /usr/local/lib/ruby/2.7.0 -r
  | ./siteconf20221123-60-1judedp.rb extconf.rb
  |  
  | +------------------------------------------------------------------------------+
  | \| ** Preparing to build the ddtrace profiling native extension... **           \|
  | \|                                                                              \|
  | \| If you run into any failures during this step, you can set the               \|
  | \| `DD_PROFILING_NO_EXTENSION` environment variable to `true` e.g.              \|
  | \| `$ DD_PROFILING_NO_EXTENSION=true bundle install` to skip this step.         \|
  | \|                                                                              \|
  | \| If you disable this extension, the Datadog Continuous Profiler will          \|
  | \| not be available, but all other ddtrace features will work fine!             \|
  | \|                                                                              \|
  | \| If you needed to use this, please tell us why on                             \|
  | \| <https://github.com/DataDog/dd-trace-rb/issues/new> so we can fix it :)      \|
  | \|                                                                              \|
  | \| Thanks for using ddtrace! You rock!                                          \|
  | +------------------------------------------------------------------------------+
  |  
  | extconf.rb:204: warning: already initialized constant
  | MakeMakefile::COMMON_HEADERS
  | /usr/local/lib/ruby/2.7.0/mkmf.rb:2684: warning: previous definition of
  | COMMON_HEADERS was here
  | checking for RUBY_MJIT_H in rb_mjit_min_header-2.7.6.h... yes
  | extconf.rb:208: warning: already initialized constant
  | MakeMakefile::COMMON_HEADERS
  | extconf.rb:204: warning: previous definition of COMMON_HEADERS was here
  | creating extconf.h
  | creating Makefile
  |  
  | current directory:
  | /app/vendor/bundle/ruby/2.7.0/gems/ddtrace-1.6.1/ext/ddtrace_profiling_native_extension
  | make "DESTDIR=" clean
  |  
  | current directory:
  | /app/vendor/bundle/ruby/2.7.0/gems/ddtrace-1.6.1/ext/ddtrace_profiling_native_extension
  | make "DESTDIR="
  | compiling clock_id_from_pthread.c
  | compiling clock_id_noop.c
  | compiling collectors_cpu_and_wall_time.c
  | compiling collectors_cpu_and_wall_time_worker.c
  | compiling collectors_stack.c
  | collectors_stack.c: In function ‘sample_thread’:
  | collectors_stack.c:151:7: error: missing initializer for field ‘start_line’ of
  | ‘ddog_Function {aka struct ddog_Function}’ [-Werror=missing-field-initializers]
  | },
  | ^
  | In file included from
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11:0,
  | from libdatadog_helpers.h:3,
  | from collectors_stack.c:5:
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:307:11:
  | note: ‘start_line’ declared here
  | int64_t start_line;
  | ^~~~~~~~~~
  | collectors_stack.c: In function ‘maybe_add_placeholder_frames_omitted’:
  | collectors_stack.c:307:5: error: missing initializer for field ‘start_line’ of
  | ‘ddog_Function {aka struct ddog_Function}’ [-Werror=missing-field-initializers]
  | },
  | ^
  | In file included from
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11:0,
  | from libdatadog_helpers.h:3,
  | from collectors_stack.c:5:
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:307:11:
  | note: ‘start_line’ declared here
  | int64_t start_line;
  | ^~~~~~~~~~
  | collectors_stack.c: In function ‘record_placeholder_stack_in_native_code’:
  | collectors_stack.c:344:5: error: missing initializer for field ‘start_line’ of
  | ‘ddog_Function {aka struct ddog_Function}’ [-Werror=missing-field-initializers]
  | },
  | ^
  | In file included from
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11:0,
  | from libdatadog_helpers.h:3,
  | from collectors_stack.c:5:
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:307:11:
  | note: ‘start_line’ declared here
  | int64_t start_line;
  | ^~~~~~~~~~
  | collectors_stack.c: In function ‘sampling_buffer_new’:
  | collectors_stack.c:376:5: error: missing initializer for field ‘is_folded’ of
  | ‘ddog_Location {aka struct ddog_Location}’ [-Werror=missing-field-initializers]
  | buffer->locations[i] = (ddog_Location) {.lines = (ddog_Slice_line) {.ptr =
  | &buffer->lines[i], .len = 1}};
  | ^~~~~~
  | In file included from
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11:0,
  | from libdatadog_helpers.h:3,
  | from collectors_stack.c:5:
  | /app/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:360:8:
  | note: ‘is_folded’ declared here
  | bool is_folded;
  | ^~~~~~~~~
  | cc1: all warnings being treated as errors
  | Makefile:244: recipe for target 'collectors_stack.o' failed
  | make: *** [collectors_stack.o] Error 1
  |  
  | make failed, exit code 2
  |  
  | Gem files will remain installed in
  | /app/vendor/bundle/ruby/2.7.0/gems/ddtrace-1.6.1 for inspection.
  | Results logged to
  | /app/vendor/bundle/ruby/2.7.0/extensions/x86_64-linux/2.7.0/ddtrace-1.6.1/gem_make.out
  |  
  | An error occurred while installing ddtrace (1.6.1), and Bundler cannot continue.
  | Make sure that `gem install ddtrace -v '1.6.1'` succeeds before bundling.

```

**Environment**

* **ddtrace version:** 1.6
* **Configuration block (`Datadog.configure ...`):** let me know if this is relevant.. this happens during bundle install
* **Ruby version:** 2.7.6
* **Operating system:** Ubuntu 18.04 in Docker 20.10.14
* **Relevant library versions:**
  * gcc 7.5.0
  * bundler 2.1.4
  *  gems
     ```
        libdatadog (0.9.0.1.0)
        libddwaf (1.5.1.0.0)
       dogstatsd-ruby (5.5.0)
        ffi (1.15.5)
        google-protobuf (3.19.4)
      ```

**Comment by TonyCTHsu** [comment]
👋 @zach-taylor Have you tried follow [this](https://github.com/DataDog/dd-trace-rb/issues/2388#issuecomment-1322408831) to mitigate?

**Comment by zach-taylor** [comment]
@TonyCTHsu yep, that seems to work. We are using the profiler though, so we would like to assist in fixing.

**Comment by zach-taylor** [comment]
@TonyCTHsu ah, I see the issue has been fixed, but not yet released. Apologies!

**Comment by TonyCTHsu** [comment]
@zach-taylor No worries, sorry for causing the issue. We will make sure to notify you once it is released.

**Comment by ivoanjo** [comment]
@zach-taylor apologies for the issue and thanks for the report! This was caused by an incorrect warning on the compiler, together with the "turn-warnings-into-errors" option that should only be enabled on our CI, but ended up being enabled by mistake on other setups.

Both are fixed for next release: https://github.com/DataDog/dd-trace-rb/pull/2425 changes the code to avoid triggering the incorrect warning, and https://github.com/DataDog/dd-trace-rb/pull/2378 makes sure the warnings-into-errors settings only happens in our CI.

If you need a more targeted workaround, changing your setup to run `CI=false bundle install` and removing the previous `DD_PROFILING_NO_EXTENSION` should allow the profiler to still build correctly. 

We'll leave a note here once the next release is out. Hopefully this works for you, but do please let us know if it doesn't! (Or if there's anything we can help with!)

**Comment by ivoanjo** [comment]
Thanks for the patience, everyone! The fix has now been released as part of [version 1.7.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v1.7.0)

**Comment by ivoanjo** [comment]
(As always, let us know how we can help and keep the feedback coming! 😄 )
